torch/distributions/multinomial.py

```diff
@@ -101,7 +101,8 @@ def sample(self, sample_shape=torch.Size()):
     def log_prob(self, value):
         if self._validate_args:
             self._validate_sample(value)
-        logits, value = broadcast_all(self.logits.clone(memory_format=torch.contiguous_format), value)
+        logits, value = broadcast_all(self.logits, value)
```

Comment:
Why did this have to change?

Reply:
This was a bug exposed by the new checks in `masked_fill`. It first broadcasts `self.logits` with `value`, then writes to the broadcasted (i.e. zero-strided) array in a way that overwrites entire rows instead of only where `value == 0`.